Can't compare chars

Hi,

Love this library. I have been using a years-old version with no issues for some time now. I tried to update and ran into this linker error:

```
FAILED: fishtank-tests
: && /usr/bin/c++   CMakeFiles/fishes-tests.dir/fish-test.cpp.o CMakeFiles/fishes-tests.dir/tank-test.cpp.o CMakeFiles/fishes-tests.dir/string-screen-test.cpp.o CMakeFiles/fishes-tests.dir/shape-test.cpp.o CMakeFiles/fishtank-tests.dir/tests.cpp.o -o fishtank-tests  -Wl,-rpath,"\$ORIGIN"  libfishes.so && :
/usr/bin/ld: CMakeFiles/fishes-tests.dir/string-screen-test.cpp.o: in function `void utest_type_printer<char>(char)':
string-screen-test.cpp:(.text._Z18utest_type_printerIcEvT_[_Z18utest_type_printerIcEvT_]+0x18): undefined reference to `utest_type_deducer<char, false>::_(char)'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

I minimized the repro to this small C++ snippet:

```c++
#include "utest.h"

char a = 'c';

UTEST(ExampleTest, Test1) {
    EXPECT_EQ(a, 'c');
}

UTEST_MAIN();
```

which I put in [this repo](https://github.com/tekknolagi/utest-repro)

If I add this snippet:

```c++
template <> struct utest_type_deducer<char, false> {
  static void _(const char c) {
    UTEST_PRINTF("%d", static_cast<int>(c));
  }
};
```

then the linker error goes away, but then some other (maybe related?) issue with string comparisons crops up. I'll try to debug further, but this seems to be an issue anyway.

--- a/test/test.c
+++ b/test/test.c
@@ -169,6 +169,13 @@ UTEST(c, LongDouble) {
 }
 
 UTEST(c, Char) {
+  char a = 1;
+  char b = 2;
+  EXPECT_NE(a, b);
+  ASSERT_NE(a, b);
+}
+
+UTEST(c, SignedChar) {
   signed char a = 1;
   signed char b = 2;
   EXPECT_NE(a, b);

--- a/test/test.cpp
+++ b/test/test.cpp
@@ -166,6 +166,13 @@ UTEST(cpp, LongDouble) {
 }
 
 UTEST(cpp, Char) {
+  char a = 1;
+  char b = 2;
+  EXPECT_NE(a, b);
+  ASSERT_NE(a, b);
+}
+
+UTEST(cpp, SignedChar) {
   signed char a = 1;
   signed char b = 2;
   EXPECT_NE(a, b);

--- a/test/test11.c
+++ b/test/test11.c
@@ -161,6 +161,13 @@ UTEST(c11, LongDouble) {
 }
 
 UTEST(c11, Char) {
+  char a = 1;
+  char b = 2;
+  EXPECT_NE(a, b);
+  ASSERT_NE(a, b);
+}
+
+UTEST(c11, SignedChar) {
   signed char a = 1;
   signed char b = 2;
   EXPECT_NE(a, b);

--- a/test/test11.cpp
+++ b/test/test11.cpp
@@ -173,6 +173,13 @@ UTEST(cpp11, LongDouble) {
 }
 
 UTEST(cpp11, Char) {
+  char a = 1;
+  char b = 2;
+  EXPECT_NE(a, b);
+  ASSERT_NE(a, b);
+}
+
+UTEST(cpp11, SignedChar) {
   signed char a = 1;
   signed char b = 2;
   EXPECT_NE(a, b);

--- a/test/test14.cpp
+++ b/test/test14.cpp
@@ -173,6 +173,13 @@ UTEST(cpp14, LongDouble) {
 }
 
 UTEST(cpp14, Char) {
+  char a = 1;
+  char b = 2;
+  EXPECT_NE(a, b);
+  ASSERT_NE(a, b);
+}
+
+UTEST(cpp14, SignedChar) {
   signed char a = 1;
   signed char b = 2;
   EXPECT_NE(a, b);

--- a/test/test17.cpp
+++ b/test/test17.cpp
@@ -173,6 +173,13 @@ UTEST(cpp17, LongDouble) {
 }
 
 UTEST(cpp17, Char) {
+  char a = 1;
+  char b = 2;
+  EXPECT_NE(a, b);
+  ASSERT_NE(a, b);
+}
+
+UTEST(cpp17, SignedChar) {
   signed char a = 1;
   signed char b = 2;
   EXPECT_NE(a, b);

--- a/test/test99.c
+++ b/test/test99.c
@@ -161,6 +161,13 @@ UTEST(c99, LongDouble) {
 }
 
 UTEST(c99, Char) {
+  char a = 1;
+  char b = 2;
+  EXPECT_NE(a, b);
+  ASSERT_NE(a, b);
+}
+
+UTEST(c99, SignedChar) {
   signed char a = 1;
   signed char b = 2;
   EXPECT_NE(a, b);

--- a/utest.h
+++ b/utest.h
@@ -447,6 +447,15 @@ struct utest_type_deducer final {
   static void _(const T t);
 };
 
+template <> struct utest_type_deducer<char, false> {
+  static void _(const char c) {
+    if (std::is_signed<decltype(c)>::value) {
+      UTEST_PRINTF("%d", static_cast<int>(c));
+    } else {
+      UTEST_PRINTF("%u", static_cast<unsigned int>(c));
+    }
+  }
+};
 template <> struct utest_type_deducer<signed char, false> {
   static void _(const signed char c) {
     UTEST_PRINTF("%d", static_cast<int>(c));

